flexget/plugins/search_t411.py

```diff
@@ -0,0 +1,275 @@
+              VFQ, VFSTFR, VOSTFR, VOASTA
+
+              BDrip-BRrip-SD, Bluray-4K, Bluray-Full-Remux, DVD-R-5,
+              DVD-R-9, DVDrip, HDrip-1080p, HDrip-720p, HDlight-1080p,
+              HDlight-720p, TVrip-SD, TVripHD-1080p, TVripHD-720p,
+              VCD-SVCD-VHSrip, WEBrip, WEBripHD-1080p, WEBripHD-1080p
+
+              2D, 3D-Converti-Amateur, 3D-Converti-Pro, 3D-Natif
+    """
+    schema = {
+        'type': 'object',
+        'properties': {
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'category': {'type': 'string'},
+            'sub_category': one_or_more(
+                {'type': 'string', 'enum': list(SUB_CATEGORIES)}
+            ),
+        },
+        'required': ['username', 'password'],
+        'additionalProperties': False
+    }
+
+    @plugin.internet(log)
+    def search(self, task, entry, config=None):
+        """
+        Search torrent in private torrent tracker t411
+        """
+
+        if not isinstance(config, dict):
+            config = {}
+
+        category = config.get('category')
+
+        if category == 'Serie-TV':
+            sub_categories_dict = SUB_CATEGORIES.copy()
+            sub_categories_dict.update(SUB_CATEGORIES_SERIES)
+
+        if category in list(CATEGORIES):
+            category = CATEGORIES[category]
+
+        sub_categories = config.get('sub_category')
+        if not isinstance(sub_categories, list):
+            sub_categories = [sub_categories]
+
+        filter_url = ''
+        if isinstance(category, int):
+            filter_url = '?cat=%s' % str(category)
+
+            if sub_categories[0] is not None:
+                sub_categories = [sub_categories_dict[c] for c in sub_categories]
+                filter_url = filter_url + '&' + '&'.join([quote_plus('term[%s][]' % c[0]).
+                                                          encode('utf-8') + '=' + str(c[1])
+                                                          for c in sub_categories if c != 0])
+        entries = set()
+        auth_handler = t411Auth(config.get('username'), config.get('password'))
+        for search_string in entry.get('search_strings', [entry['title']]):
+            query = normalize_unicode(search_string)
+            url_search = ('/torrents/search/' +
+                          quote_plus(query.encode('utf-8')) +
+                          filter_url)
+
+            response = requests.get(T411_BASE_URL + url_search, auth=auth_handler)
+            res_json = response.json()
+
+            if 'errors' in res_json.keys():
+                auth_handler = t411Auth(config.get('username'), config.get('password'), force_auth=True)
+                res_json = requests.get(T411_BASE_URL + url_search, auth=auth_handler).json()
+
+                if 'errors' in res_json.keys():
+                    raise plugin.PluginError(res['error'])
+
+            for torrent in res_json['torrents']:
+                entry = Entry()
```

Comment:
I think it's bad practice to overwrite function arguments -- I know a few of the old search plugins do it, but new ones shouldn't. Python does not have local scopes like in C.